proxy/grpcproxy/watcher_single.go

```diff
@@ -53,9 +52,8 @@ func (ws watcherSingle) run() {
 	defer close(ws.donec)
 
 	for wr := range ws.ch {
-		ws.rev = wr.Header.Revision
+		ws.lastSeenRev = wr.Header.Revision
```

Comment:
This will drop events. Set a watcher at rev=10 and have a store at rev=100. Receive rev=11 and it could coalesce on rev=100, missing events on revs 12 to 100.